azure_img_utils/azure_image.py

```diff
@@ -604,27 +634,69 @@ def go_live_with_offer(
 
         Returns the operation uri.
         """
-        return go_live_with_cloud_partner_offer(
-            self.access_token,
+        endpoint = get_cloud_partner_endpoint(
             offer_id,
-            publisher_id
+            publisher_id,
+            go_live=True
+        )
+        headers = get_cloud_partner_api_headers(
+            self.access_token,
+            content_type='application/json'
+        )
+
+        response = process_request(
+            endpoint,
+            headers,
+            method='post',
+            json_response=False
         )
 
+        return response.headers['Location']
+
     def get_offer_status(self, offer_id, publisher_id) -> str:
         """
         Returns the status of the offer.
         """
-        return get_cloud_partner_offer_status(
-            self.access_token,
+        endpoint = get_cloud_partner_endpoint(
             offer_id,
-            publisher_id
+            publisher_id,
+            status=True
         )
+        headers = get_cloud_partner_api_headers(self.access_token)
+
+        response = process_request(
+            endpoint,
+            headers,
+            method='get'
+        )
+
+        status = response.get('status', 'unkown')
+
+        if status == 'running':
+            signoff_status = jmespath.search(
+                "steps[?stepName=='publisher-signoff'].status | [0]",
+                response
+            )
+        if signoff_status == 'waitingForPublisherReview':
```

Comment:
Not sure if this is new or existing bug but signoff_status will not be defined if status is not "running".

Reply:
Good catch, Sean!
It was a bug in the refactor. The if for the signoff_status is expected to go inside the outer if.
Fixed!